src/main/java/edu/cornell/kfs/paymentworks/service/impl/PaymentWorksKfsServiceImpl.java

```diff
@@ -57,394 +52,314 @@
-
-				// create vendor for edit
-				VendorDetail vendorDetail = new PaymentWorksVendorUpdateConversionUtil()
-						.createVendorDetailForEdit(newVendor, oldVendor, paymentWorksVendor);
-
-				// Route Maintenance document
-				MaintenanceDocument vendorMaintDoc = null;
-				try {
-					vendorMaintDoc = (MaintenanceDocument) documentService.getNewDocument("PVEN");
-					vendorMaintDoc.getDocumentHeader()
-							.setDocumentDescription(StringUtils.defaultString(paymentWorksVendor.getVendorName()));
-					vendorMaintDoc.getOldMaintainableObject().setBusinessObject(oldVendor);
-					vendorMaintDoc.getNewMaintainableObject().setBusinessObject(vendorDetail);
-					vendorMaintDoc.getNewMaintainableObject()
-							.setMaintenanceAction(KFSConstants.MAINTENANCE_EDIT_ACTION);
-					vendorMaintDoc.getNewMaintainableObject().setDocumentNumber(vendorMaintDoc.getDocumentNumber());
-
-					boolean isVendorLocked = checkForLockingDocument(vendorMaintDoc);
-
-					if (!isVendorLocked) {
-						boolean isValid = true;
-						try {
-							vendorMaintDoc.validateBusinessRules(new RouteDocumentEvent(vendorMaintDoc));
-						} catch (ValidationException ve) {
-							LOG.error("Failed to route Vendor Edit document due to business rule error(s): "
-									+ new PaymentWorksUtil().getAutoPopulatingErrorMessages(
-											GlobalVariables.getMessageMap().getErrorMessages()));
-							isValid = false;
-						}
-
-						if (isValid) {
-							// add note
-							// addNoteForNewVendorResponsesToVendor(vendorMaintDoc.getNewMaintainableObject());
-
-							// route document
-							documentService.routeDocument(vendorMaintDoc, null, null);
-
-							isRouted = true;
-
-							// add document number
-							documentNumbers.append(vendorMaintDoc.getDocumentNumber() + ",");
-						}
-					} else {
-						LOG.error("Failed to route Vendor Edit document due to lock");
-						GlobalVariables.getMessageMap().putError("vendorDetail", KFSKeyConstants.ERROR_CUSTOM,
-								"Failed to route Vendor Edit document due to lock");
-						isRouted = false;
-					}
-				} catch (Exception e) {
-					LOG.error("Failed to route Vendor document due to error(s): " + e.getMessage());
-					isRouted = false;
-				}
+				routed = processNonDuplicateVendorDetail(paymentWorksVendor, documentNumbers, oldVendor, newVendor);
 			} else {
-				GlobalVariables.getMessageMap().putError("vendorDetail", KFSKeyConstants.ERROR_CUSTOM,
-						"Duplicate vendor update detected");
-				isRouted = false;
+				GlobalVariables.getMessageMap().putError("vendorDetail", KFSKeyConstants.ERROR_CUSTOM, "Duplicate vendor update detected");
+				routed = false;
 			}
 		}
 
 		if (documentNumbers.length() > 0) {
-			paymentWorksVendor
-					.setDocumentNumberList(documentNumbers.toString().substring(0, documentNumbers.length() - 1));
+			paymentWorksVendor.setDocumentNumberList(documentNumbers.toString().substring(0, documentNumbers.length() - 1));
+		}
+		return routed;
```

Comment:
I believe we will want that result, but we will find out for certain when we turn on the edit vendor end point.  Does that sound OK?